layerindex/tools/import_otherdistro.py

```diff
@@ -0,0 +1,789 @@
+            return 1, None
+    else:
+        layer = LayerItem()
+        layer.name = args.layer
+        layer.layer_type = 'M'
+        layer.summary = layer.name
+        layer.description = layer.name
+        layer.comparison = True
+        layer.save()
+
+    layerbranch = layer.get_layerbranch(args.branch)
+    if not layerbranch:
+        layerbranch = LayerBranch()
+        layerbranch.layer = layer
+        layerbranch.branch = branch
+        layerbranch.save()
+
+    return 0, layerbranch
+
+
+def get_update_obj(args):
+    from layerindex.models import Update
+    updateobj = None
+    if args.update:
+        updateobj = Update.objects.filter(id=int(args.update))
+        if not updateobj:
+            logger.error("Specified update id %s does not exist in database" % args.update)
+            sys.exit(1)
+        updateobj = updateobj.first()
+    return updateobj
+
+
+def import_specdir(metapath, layerbranch, existing, updateobj, pwriter):
+    dirlist = os.listdir(metapath)
+    total = len(dirlist)
+    speccount = 0
+    for count, entry in enumerate(dirlist):
+        if os.path.exists(os.path.join(metapath, entry, 'dead.package')):
+            logger.info('Skipping dead package %s' % entry)
+            continue
+        specfiles = glob.glob(os.path.join(metapath, entry, '*.spec'))
+        if specfiles:
+            import_specfiles(specfiles, layerbranch, existing, updateobj, metapath)
+            speccount += len(specfiles)
+        else:
+            logger.warn('Missing spec file in %s' % os.path.join(metapath, entry))
+        if pwriter:
+            pwriter.write(int(count / total * 100))
+    return speccount
+
+
+def import_specfiles(specfiles, layerbranch, existing, updateobj, reldir):
+    from layerindex.models import ClassicRecipe, ComparisonRecipeUpdate
+    recipes = []
+    for specfile in specfiles:
+        specfn = os.path.basename(specfile)
+        specpath = os.path.relpath(os.path.dirname(specfile), reldir)
+        recipe, created = ClassicRecipe.objects.get_or_create(layerbranch=layerbranch, filepath=specpath, filename=specfn)
+        if created:
+            logger.info('Importing %s' % specfn)
+        elif recipe.deleted:
+            logger.info('Restoring and updating %s' % specpath)
+            recipe.deleted = False
+        else:
+            logger.info('Updating %s' % specpath)
+        recipe.layerbranch = layerbranch
+        recipe.filename = specfn
+        recipe.filepath = specpath
+        update_recipe_file(specfile, recipe, reldir)
+        recipe.save()
+        existingentry = (specpath, specfn)
+        if existingentry in existing:
+            existing.remove(existingentry)
+        if updateobj:
+            rupdate, _ = ComparisonRecipeUpdate.objects.get_or_create(update=updateobj, recipe=recipe)
+            rupdate.meta_updated = True
+            rupdate.save()
+        recipes.append(recipe)
+    return recipes
+
+
+def import_pkgspec(args):
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This function calls 17 other functions. By comparison, 98% of the functions in the CodeGuru reference dataset use fewer. This indicates the function is highly coupled with other functions. A function that is highly coupled with other functions is difficult to understand and its behavior might change unexpectedly when one of its referenced functions is updated. High coupling could also increase the integration test complexity, maintenance cost and technical debt. We recommend that you simplify this function or break it into multiple functions.